frontend/components/Vehicles/VehicleDialog.vue

```diff
@@ -0,0 +1,76 @@
+<template>
+  <v-dialog v-model="showDialog" max-width="500px">
+    <template v-slot:activator="{ on, attrs }">
+      <v-btn color="primary" dark v-bind="attrs" v-on="on">
+        Add vehicle
+      </v-btn>
+    </template>
+    <v-card>
+      <v-card-title>
+        <span class="text-h5">Add new vehicle</span>
+      </v-card-title>
+      <v-card-text>
+        <v-container>
+          <v-row>
+            <v-col cols="12" sm="6" md="4">
+              <v-text-field v-model="editedItem.make" label="Make" single-line required></v-text-field>
+            </v-col>
+            <v-col cols="12" sm="6" md="4">
+              <v-text-field v-model="editedItem.model" label="Model" single-line required></v-text-field>
+            </v-col>
+            <v-col cols="12" sm="6" md="4">
+              <v-text-field v-model="editedItem.year" label="Year" single-line required type="number">
+              </v-text-field>
+            </v-col>
+          </v-row>
+        </v-container>
+      </v-card-text>
+      <v-card-actions>
+        <v-spacer></v-spacer>
+        <v-btn color="blue darken-1" text @click="close">
+          Cancel
+        </v-btn>
+        <v-btn color="blue darken-1" text @click="add">
+          Add
+        </v-btn>
+      </v-card-actions>
+    </v-card>
+  </v-dialog>
+</template>
+
+<script>
+export default {
+  name: 'VehicleDialog',
+
+  data: () => ({
+    showDialog: false,
+    editedItem: { make: "", model: "", year: null }
+  }),
+
+  methods: {
+    defaultItem() {
+      return { make: "", model: "", year: null };
+    },
+
+    close() {
+      this.showDialog = false;
+      this.editedItem = this.defaultItem();
+    },
+
+    add() {
+      const newItem = {
+        make: this.editedItem.make,
+        model: this.editedItem.model,
+        year: this.editedItem.year
+      }
+
+      this.$emit("createVehicle", newItem);
```

Comment:
```suggestion
      this.$emit('create', this.editedItem);
```
Use `dash-case` for the event name. Also, I noticed that you are using double quotes for strings, eslint does not warn you? Maybe you're missing one of the eslint packages such as `eslint-config-semistandard` or `eslint-config-standard`.

Reply:
done. it does warn but i used croatian keyboard so it is easier to use " " instead of ' '... i fixed those lints though through this commit.